dizoo/classic_control/pendulum/envs/pendulum_env.py

```diff
@@ -58,12 +61,17 @@ def seed(self, seed: int, dynamic_seed: bool = True) -> None:
 
     def step(self, action: np.ndarray) -> BaseEnvTimestep:
         assert isinstance(action, np.ndarray), type(action)
+        # if require discrete env, convert actions to [-2 ~ 2] float actions
+        if not self._is_continue:
+            action=(action-(self._discrete_action_num-1)/2)/((self._discrete_action_num-1)/4)
```

Comment:
you need to transform it to [-1, 1] here, then `affine_transform` will  make it to [-2, 2]